imgix/contentful
# feat(upload): create upload modal

## Description

This PR creates the upload modal in the `<Dialog />` component. This modal is displayed whenever the user clicks on `Upload` in the header. The modal allows the user to

1. Change the destination source
2. Change the destination path
3. Preview the asset before upload


This PR also adds `uploadForm` to state.

```typescript
  uploadForm: {
    file?: any;
    source?: {
      id?: string;
      name?: string;
      domain?: string;
    };
    previewSource?: string;
  };
```

This PR also adds some helper class methods to manage state.

```js
setIsUploading = (value: boolean) => {
    this.setState({ isUploading: value });
  };

  setUploadSource = (source: { id: string; name: string; domain: string }) => {
    const uploadForm = { ...this.state.uploadForm, source };
    this.setState({ uploadForm });
  };

  updateFileForm = (file: any, previewSource: string, isUploading: boolean) => {
    const uploadForm = {
      ...this.state.uploadForm,
      file,
      source: this.state.selectedSource,
      previewSource,
    };
    this.setState({ uploadForm, isUploading });
  };
```

A future PR should add a callback on upload confirmation that actually sends the file to the imgix asset manager via then management SDK.




# Before
No upload modal.


# After
The upload modal is displayed when the user clicks upload



# Steps to test
- pull down the branch
- run `npm run start`
- open contentful staging env
- click "add image"
- click "upload"
- see modal open

# 🖼️  Screenshots

https://user-images.githubusercontent.com/16711614/198851677-a5c5b739-6af3-4218-ad4d-a47bf856d62f.mov



## Checklist
